ibis/expr/tests/test_newrels.py

```diff
@@ -1191,3 +1205,46 @@ def test_name_collisions_raise():
     abc = a.join(b, [a.a == b.a]).join(c, [a.a == c.a])
     with pytest.raises(IntegrityError):
         abc.filter(abc.a < 1)
+
+
+def test_self_view_join_followed_by_aggregate_correctly_dereference_fields():
```

Comment:
This is the test case covering the issue, the other test changes were just required to conform with the new IR.

Reply:
Verified this by inspecting the IR repr, it looks good now!

```
(Pdb) pp expected_agg
r0 := UnboundTable: t
  a int64
  b int64
  f int64
  g string

r1 := Aggregate[r0]
  groups:
    g: r0.g
    a: r0.a
    b: r0.b
  metrics:
    total: Sum(r0.f)

r2 := SelfReference[r1]

r3 := SelfReference[r1]

r4 := JoinChain[r2]
  JoinLink[inner, r3]
    r2.a == r3.b
  values:
    g:           r2.g
    a:           r2.a
    b:           r2.b
    total:       r2.total
    g_right:     r3.g
    a_right:     r3.a
    b_right:     r3.b
    total_right: r3.total

Aggregate[r4]
  groups:
    g: r4.g
  metrics:
    metric: Max(r4.total - r4.total_right)
```